handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -934,7 +934,15 @@ private boolean wrapNonAppData(ChannelHandlerContext ctx, boolean inUnwrap) thro
                 SSLEngineResult result = wrap(alloc, engine, Unpooled.EMPTY_BUFFER, out);
 
                 if (result.bytesProduced() > 0) {
-                    ctx.write(out);
+                    ctx.write(out).addListener(new ChannelFutureListener() {
```

Comment:
will this add much overhead?

Reply:
I didn't benchmark it but I don't anticipate a single future listener to add much. This is limited to "control" data and the impacts of keeping resources around until the handshake timeout fires are also non-zero.